src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
-    parameters = ['index', 'toil.bam']
-    inputs=['toil.bam']
-    outputs={'toil.bam.bai': None}
-    docker_call(work_dir = work_dir,
-                parameters = parameters,
-                tool = 'quay.io/ucsc_cgl/samtools',
-                inputs=inputs,
-                outputs=outputs)
-    # Update FileStore and call child
-    shared_ids['toil.bam.bai'] = job.fileStore.writeGlobalFile(output_path)
-    job.addChildJobFn(haplotype_caller, shared_ids, input_args, cores = input_args.cpu_count)
-
-
-def haplotype_caller(job, shared_ids, input_args):
+        raise ValueError('Accepted formats: FASTQ or BAM\n'
+                         'Could not determine {} file type from URL:\n{}'.format(uuid, url))
+
+    rm_secondary = job.wrapJobFn(run_samtools_view,
+                                 get_bam.rv(),
+                                 flag='0x800',
+                                 ncores=cores,
+                                 disk=config.file_size, cores=cores)
+
+    sort_bam_disk = PromisedRequirement(lambda x: 3*x.size + human2bytes('10G'), rm_secondary.rv())
+    sort_bam = job.wrapJobFn(run_samtools_sort,
+                             rm_secondary.rv(),
+                             ncores=cores,
+                             disk=sort_bam_disk)
+
+    index_bam_disk = PromisedRequirement(lambda x: int(1.5 * x.size), sort_bam.rv())
+    index_bam = job.wrapJobFn(run_samtools_index, sort_bam.rv(), disk=index_bam_disk)
+
+    job.addChild(get_bam)
+    get_bam.addChild(rm_secondary)
+    rm_secondary.addChild(sort_bam)
+    sort_bam.addChild(index_bam)
+    return sort_bam.rv(), index_bam.rv()
+
+
+def setup_and_run_bwa_kit(job, url, config):
+    """
+    Downloads and aligns paired FASTQs to reference genome using BWA.
+
+    :param JobFunctionWrapping Job: passed by Toil
+    :param str url: FASTQ sample URL
+    :param Namespace config: Input parameters and shared FileStoreIDs
+    :return: BAM FileStoreID
+    :rtype: str
+    """
+    job.fileStore.logToMaster("Aligning sample: {}".format(config.uuid))
+    cores = multiprocessing.cpu_count()
+    # Download fastq files
+    fq1 = job.addChildJobFn(download_url_job,
+                            url,
+                            name='toil.1.fq',
+                            s3_key_path=config.ssec,
+                            disk='50G')
+    # Assumes second fastq url is identical
+    fq2_url = url.replace('1.fq', '2.fq')
+    fq2 = job.addChildJobFn(download_url_job,
+                            fq2_url,
+                            name='toil.2.fq',
+                            s3_key_path=config.ssec,
+                            disk='50G')
+
+    bwa_config = deepcopy(config)
+    bwa_config.r1 = fq1.rv()
+    bwa_config.r2 = fq2.rv()
+    # bwa_alignment uses a different naming convention
+    bwa_config.ref = config.genome_fasta
+    bwa_config.fai = config.genome_fai
+    bwakit_disk = PromisedRequirement(lambda x, y: 3*(x.size + y.size) + human2bytes('8G'),
```

Comment:
Where ever a constant value (e.g., the '8G' in this line) comes in, you need to document where this constant comes from.